### ISSUE TITLE
Support `FixedSizeListArray` for array

### ISSUE THREAD

**Issue by Weijun-H** [apache/datafusion#8249]
### Is your feature request related to a problem or challenge?

The array function currently does not support `FixedSizeList`, and this ticket is to track its progress.
- [ ] `array_length `
- [ ] `array_element`
- [ ] `array_slice`
- [ ] `array_pop_back`
- [ ] `array_pop_front`
- [ ] `as_list_array`
- [ ] `array_prepend`
- [ ] `array_empty`
- [ ] `array_repeat`
- [ ] `array_position`
- [ ] `array_positions`
- [ ] `array_replace`, `array_replace_n`, `array_replace_all`
- [ ] `cardinality`
- [ ] `array_dims`, `array_ndims`
- [ ] `array_has`, `array_has_any`, `array_has_all`
- [ ] `array_intersect`

**Comment by jayzhan211** [comment]
@Weijun-H I thought fixedsizelist is all handled since we convert it to List.

**Comment by Weijun-H** [comment]
> @Weijun-H I thought fixedsizelist is all handled since we convert it to List.
> 
> If we can convert fixedsizelist to list in early stage, we can significant reduce many fixedsizelist handling
> 
> However, #8902 is still useful, so we can easily test fixedsizelist with `arrow_cast`

@jayzhan211 I totally agree with you. As we convert `FixedSizeList` to `List`, we could not consider it on function.

**Comment by Jefffrey** [comment]
I plan to go through the array functions in datafusion-functions-nested and check this support as well as tests

Edit: I'll track progress it in the same comment as issue #8185 here: https://github.com/apache/datafusion/issues/8185#issuecomment-3310789742

**Comment by Jefffrey** [comment]
Per this comment https://github.com/apache/datafusion/issues/8185#issuecomment-3319160384 I'll close this as done too
